src/main/webapp/script.js

```diff
@@ -47,22 +51,80 @@ function initializeRow(json, key, currentRow) {
   let index = 0;
   for (const type of json[key]) {
     const columnEntry = document.createElement('td');
-    console.log(type);
-    console.log(type.toUpperCase());
+
     if ((type.toUpperCase() === 'POSE' && index === 0) ||
       (type.toUpperCase() === 'POINTCLOUD' && index === 1)) {
       const link = document.createElement('a');
       link.href = '/getrun?id=' + key + '&dataType=' + json[key];
       link.innerText = type;
       columnEntry.appendChild(link);
+      currentRow.appendChild(columnEntry);
+      /* Makes checkboxes for each data entry. */
+      const checkbox = document.createElement('INPUT');
+      checkbox.type = 'checkbox';
+      checkbox.setAttribute('name', 'check');
+      checkbox.setAttribute('value', link.getAttribute('name'));
+      currentRow.appendChild(checkbox);
     } else {
       columnEntry.innerText = '';
+      currentRow.appendChild(columnEntry);
     }
-    currentRow.appendChild(columnEntry);
     index++;
   }
   return currentRow;
 }
+
+/* Select all Checkboxes function. */
+function toggle(source) {
+  checkboxes = document.getElementsByName('check');
+  for (let i=0, n=checkboxes.length; i<n; i++) {
+    checkboxes[i].checked = source.checked;
+  }
+}
+
+/*
+ * Make url for multiple runs selected. should be in the style of
+ ** link.href = "/home.html?id=" + key + "&dataType=" + json[key]
+ */
+function makeMultiRunURL() {
+  const checkboxes = document.getElementsByName('check');
+  let count =0;
+  const link = document.createElement('a');
+  link.href = '/home.html?';
+
+  for (let i=0, n=checkboxes.length; i<n; i++) {
```

Comment:
Probably just separate this out into a function, also use for(box of checkboxes).

Reply:
Also you create the checkboxes so you shouldn't have to query the dom to retrieve their values. Consider storing them in a map/ list upon creation, so you can access them directly.